### PR TITLE
updates Frame to use enum instead of string types

### PR DESCRIPTION
**Pull request by dbrudner**
I'm thinking there's a better way to implement this that doesn't involve exporting/import `TYPE`, but I don't know what that would be.

With a static property as suggested in the ticket, I wasn't able to figure how to use that outside of the class when creating an instance of Frame (i.e., `const frame = new Frame({ type: Frame.TYPES.IFRAME });` or something like `Frame({ type: this.constructor.TYPES.IFRAME });` or `Frame({ type: this.TYPES.IFRAME });`

If I'm missing something, I'm open to changing anything.

### CODE REVIEW

File: lib/recurly/frame.js [hunk 1]
```diff
@@ -31,6 +31,14 @@ export function factory (options) {
   return new Frame(options);
 }
 
+/**
+ * @enum {Number}
+ */
+export const TYPE = {
+  WINDOW: 0,
+  IFRAME: 1
+};
```
Comment (chrissrogers) [inline lib/recurly/frame.js]: I'm thinking we can put this onto a static property on `Frame`

```
class Frame {
  static TYPES = {
    WINDOW: 0,
    IFRAME: 1
  };
}
```

File: lib/recurly/frame.js [hunk 2]
```diff
@@ -40,7 +48,7 @@ class Frame extends Emitter {
   static DEFAULTS = {
     height: 535,
     width: 450,
-    type: 'window'
+    type: TYPE.WINDOW
```
Comment (chrissrogers) [inline lib/recurly/frame.js]: We'd then use the reference `Frame.TYPES.WINDOW`

File: lib/recurly/frame.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ const debug = require('debug')('recurly:frame');
  * @property {Recurly} recurly
  * @property {String} path            API endpoint to load
  * @property {Object} payload         Message payload to deliver to endpoint
- * @property {String} [type='window'] one of 'window' or 'iframe'
+ * @property {TYPE} [type=0]          window or iframe
```
Comment (chrissrogers) [inline lib/recurly/frame.js]: nit: `[type=Frame.TYPES.WINDOW]`

File: lib/recurly/risk/three-d-secure/strategy/adyen.js [hunk 4]
```diff
@@ -133,7 +134,7 @@ export default class AdyenStrategy extends ThreeDSecureStrategy {
       redirect_url: adyenRedirectParams.url,
       ...adyenRedirectParams.data
     };
-    this.frame = recurly.Frame({ type: 'iframe', path: '/three_d_secure/start', payload, container })
+    this.frame = recurly.Frame({ type: TYPE.IFRAME, path: '/three_d_secure/start', payload, container })
```
Comment (chrissrogers) [inline lib/recurly/risk/three-d-secure/strategy/adyen.js]: Moving the enum will obviate the need to import `TYPES` separately

```suggestion
    this.frame = recurly.Frame({ type: Frame.TYPES.IFRAME, path: '/three_d_secure/start', payload, container })
```

Comment (dbrudner) [reply lib/recurly/risk/three-d-secure/strategy/adyen.js]: This was my initial implementation, but I'm getting a reference error with this in my test: 

image

image


I tried a few other ways to access TYPES, but none of them worked. Any ideas?

Comment (chrissrogers) [reply lib/recurly/risk/three-d-secure/strategy/adyen.js]: Oh right, we'd just need to import Frame into the `adyen.js` module

```js
import Frame from '../../../frame';
```

or reference it from the recurly instance in this routine

```js
this.frame = recurly.Frame({ type: recurly.Frame.TYPES.IFRAME, path: '/three_d_secure/start', payload, container });
```

**Pull request merged**